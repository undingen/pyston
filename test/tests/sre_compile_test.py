import sre_compile
import sre_constants

r = sre_compile.compile("a(b+)c", 0)
print r.match("")
print r.match("ac")
print r.match("abc").groups()
for i in xrange(100000):
    r.match("abbc").groups()
    if i % 10000 == 0:
        print i

def identity(o):
    return o

charset = [(sre_constants.RANGE, (128, 65535))]

# Note: _optimize_charset changed behavior + interface some time
# between 2.7.7 (our stdlib) and 2.7.9 (current cpython latest),
# so this test has a .expected file to avoid running cpython.
print sre_compile._optimize_charset(charset, identity)
