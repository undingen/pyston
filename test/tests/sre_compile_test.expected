None
None
('b',)
0
10000
20000
30000
40000
50000
60000
70000
80000
90000
[('bigcharset', [2, 16843008L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 16843009L, 0L, 0L, 0L, 0L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L, 4294967295L])]
